### ISSUE TITLE
Confusion between : train, val, test.

### ISSUE THREAD

**Issue by adrienpacifico** [mljar/mljar-supervised#573]
Hi,
I think that there are some elements that creates confusion (at least to me) between different sets in mljar. 

One example are graphs created by mljar:
![image](https://user-images.githubusercontent.com/6614121/194285188-cbcb98b5-c8cd-437f-8fe4-9868ef7057d0.png)

The "test" set from the learning curve is actually the validation set (or cross-validation set)! That might be to some extent confusing. 

It would be nice to:
- [ ] relabel results that use the word "test" when it is the validation set.
- [ ] Include an option to include a true "test set" (i.e not seen or used by any algorithms).

Such an option could be nice to have on all the metrics produced with the validation set on the test set. 

it could be done as such: 

```
automl.fit(
    X,
    y,
    sample_weight = df_train_undersampled.weight_discounting.values,
    cv=cv, 
    X_test = X_test,
    y_test = y_test
)
```

ps:
it could also be nice to give a X_val, y_val option such in addition to cv (i.e. cv seems to provide a lot more flexibility, but it is not user friendly if you just want to use a validation set.

**Comment by pplonski** [comment]
> Do you think documentation/docstring (e.g. line 90) should also be modified?

Yes, all confusing places should be updated.

> Why so? Users could have an very easy way to separate train, val and test that way?

1. I'm little afraid that it might be too complex for users.
2. User can have several test datasets, for example new test data every day, then it will be nice to just evaluate vs new data.
3. Adding `evaluate()` function might be a little simpler than extending the training function with new arguments.

**Comment by Djoels** [comment]
I too have recently stumbled on these limitations... I've been running automl while keeping a "real" test set separate.
When I fit the models some report perfect validation scores (log loss 0.0) but have very low accuracy on the test set. The actual best model is not in the list.
Has there been an evolution in this for the last couple of years?
